custom_components/babybuddy/sensor.py

```diff
@@ -1,409 +1,313 @@
+            vol.Optional(ATTR_AMOUNT): cv.positive_int,
+            vol.Optional(ATTR_NOTES): cv.string,
+        },
+        "async_add_diaper_change",
+    )
+
+    platform.async_register_entity_service(
+        "add_temperature",
+        {
+            vol.Required(ATTR_TEMPERATURE): cv.positive_float,
+            vol.Optional(ATTR_TIME, default=dt_util.now()): vol.Any(
+                cv.datetime, cv.time
             ),
-            ATTR_START: call.data.get(ATTR_START),
-            ATTR_END: call.data.get(ATTR_END),
-            ATTR_TYPE: call.data.get(ATTR_TYPE).lower(),
-            ATTR_METHOD: call.data.get(ATTR_METHOD).lower(),
-            ATTR_AMOUNT: call.data.get(ATTR_AMOUNT),
-            ATTR_NOTES: call.data.get(ATTR_NOTES),
+            vol.Optional(ATTR_NOTES): cv.string,
+        },
+        "async_add_temperature",
+    )
+    platform.async_register_entity_service(
+        "add_weight",
+        {
+            vol.Required(ATTR_WEIGHT): cv.positive_float,
+            vol.Optional(ATTR_DATE, default=dt_util.now().date()): cv.date,
+            vol.Optional(ATTR_NOTES): cv.string,
+        },
+        "async_add_weight",
+    )
+
+
+@callback
+def update_items(
+    coordinator: BabyBuddyCoordinator,
+    tracked: dict,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add new sensors for new endpoint entries."""
+    if coordinator.data is not None:
+        new_entities = []
+        for child in coordinator.data[0]:
+            if child[ATTR_ID] not in tracked:
+                tracked[child[ATTR_ID]] = BabyBuddyChildSensor(coordinator, child)
+                new_entities.append(tracked[child[ATTR_ID]])
+            for description in SENSOR_TYPES:
+                if description.key == ATTR_TIMERS:
+                    continue
+                if (
+                    coordinator.data[1][child[ATTR_ID]].get(description.key)
+                    and f"{child[ATTR_ID]}_{description.key}" not in tracked
+                ):
+                    tracked[
+                        f"{child[ATTR_ID]}_{description.key}"
+                    ] = BabyBuddyChildDataSensor(coordinator, child, description)
+                    new_entities.append(tracked[f"{child[ATTR_ID]}_{description.key}"])
+        if new_entities:
+            async_add_entities(new_entities)
+
+
+class BabyBuddySensor(CoordinatorEntity, SensorEntity):
+    """Base class for Babybuddy sensors."""
+
+    coordinator: BabyBuddyCoordinator
+
+    def __init__(self, coordinator: BabyBuddyCoordinator, child: dict) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.child = child
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, child[ATTR_ID])},
+            "default_name": f"Baby {child[ATTR_FIRST_NAME]} {child[ATTR_LAST_NAME]}",
         }
 
-        baby_buddy.entities_add(endpoint, data)
-
-    def services_notes_add(call):
```

Comment:
add back services_notes_add

Reply:
I didn't include this service because I didn't see how easy it is to write down a note in Home Assistant. Do you think service is important to include?

Reply:
I think it's important so we don't end up removing existing functionality.

Reply:
Added